From 6e531d0ec1347917515845299a38927de2b9e314 Mon Sep 17 00:00:00 2001
From: Ben Bright <ben.bright@twdb.texas.gov>
Date: Mon, 15 Aug 2022 09:42:44 -0500
Subject: [PATCH] ckanext-security middleware patch

This patch is based on the ckanext-security patch file but updated for CKAN 2.9.5
Original patch for reference only:
  https://github.com/data-govt-nz/ckanext-security/blob/master/ckanext-security.patch
---
 ckan/config/middleware/flask_app.py  | 8 +++++++-
 ckan/config/middleware/pylons_app.py | 6 +++++-
 2 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/ckan/config/middleware/flask_app.py b/ckan/config/middleware/flask_app.py
index eb9ff8155..1da1db40d 100644
--- a/ckan/config/middleware/flask_app.py
+++ b/ckan/config/middleware/flask_app.py
@@ -185,7 +185,10 @@ def make_flask_stack(conf):
             data_dir=cache_dir)
 
     app.wsgi_app = RootPathMiddleware(app.wsgi_app, session_opts)
-    app.wsgi_app = SessionMiddleware(app.wsgi_app, session_opts)
+
+    # Moved as part of ckanext-security patch - next line should be commented out
+    #app.wsgi_app = SessionMiddleware(app.wsgi_app, session_opts)
+
     app.session_interface = BeakerSessionInterface()
 
     # Add Jinja2 extensions and filters
@@ -324,6 +327,9 @@ def make_flask_stack(conf):
         who_parser.remote_user_key
     )
 
+    # Moved as part of ckanext-security patch
+    app = SessionMiddleware(app, session_opts)
+
     # Update the main CKAN config object with the Flask specific keys
     # that were set here or autogenerated
     flask_config_keys = set(flask_app.config.keys()) - set(config.keys())
diff --git a/ckan/config/middleware/pylons_app.py b/ckan/config/middleware/pylons_app.py
index 126fe21f0..ecb0a25b8 100644
--- a/ckan/config/middleware/pylons_app.py
+++ b/ckan/config/middleware/pylons_app.py
@@ -64,7 +64,8 @@ def make_pylons_stack(conf, full_stack=True, static_files=True,
     # we want to be able to retrieve the routes middleware to be able to update
     # the mapper.  We store it in the pylons config to allow this.
     config['routes.middleware'] = app
-    app = SessionMiddleware(app, config)
+    # Moved as part of ckanext-security patch - next line should be commented out
+    # app = SessionMiddleware(app, config)
     app = CacheMiddleware(app, config)
 
     # CUSTOM MIDDLEWARE HERE (filtered by error handling middlewares)
@@ -137,6 +138,9 @@ def make_pylons_stack(conf, full_stack=True, static_files=True,
         who_parser.remote_user_key
     )
 
+    # Moved as part of ckanext-security patch
+    app = SessionMiddleware(app, config)
+
     # Establish the Registry for this application
     app = RegistryManager(app, streaming=False)
 
-- 
2.25.1

